daemon/emer-daemon.c

```diff
@@ -1547,6 +1580,29 @@ emer_daemon_new (const gchar             *persistent_cache_directory,
                        NULL);
 }
 
+gboolean
+emer_daemon_reset_tracking_id (EmerDaemon  *self,
+                               GError     **error)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  if (priv->override_machine_id_path != NULL)
+    {
+      if (!write_tracking_id_file (priv->override_machine_id_path, error))
+        return FALSE;
+    }
+  else
+    {
+      if (!write_tracking_id_file (TRACKING_ID_OVERRIDE, error))
+        return FALSE;
+    }
+
+  if (priv->machine_id_provider != NULL)
+    emer_machine_id_provider_reload (priv->machine_id_provider);
+
+  return TRUE;
```

Comment:
If there were a signal that `eos-metrics-instrumentation` could use to discard all its state and re-emit the boot-time events, then I think the `ResetTrackingId` method should discard all buffered events and then emit that signal. But it's both or neither: if we discard all cached events here, we will receive the "it's demo mode!" event (which is emitted from g-i-s only after this method returns) but not the boot-time events with info like the OS version, image, hardware, etc.

Reply:
(I'm not proposing adding that signal and behaviour here – I think that's a separate task.)